src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.HostStates.cs

```diff
@@ -15,21 +15,6 @@ internal partial class DiagnosticIncrementalAnalyzer
     {
         private partial class StateManager
         {
-            public IEnumerable<StateSet> GetAllHostStateSets()
-            {
-                var analyzerReferences = _workspace.CurrentSolution.SolutionState.Analyzers.HostAnalyzerReferences;
```

Comment:
accessing workspace.CurrentSolution is *baddy bad*.  this is code that is all supposed to be snapshot based.